lib/portfolio_page/pie_chart.dart

```diff
@@ -0,0 +1,272 @@
+import 'package:equity_gals_portfolio/portfolio_page/portfolio_page.dart';
+import 'portfolio_page.dart';
+import 'package:fl_chart/fl_chart.dart';
+import 'package:flutter/material.dart';
+
+class EGPieChart extends StatefulWidget {
+  final double maxContentWidth;
+  const EGPieChart({Key? key, required this.maxContentWidth}) : super(key: key);
+
+  @override
+  State<EGPieChart> createState() => PieChartState();
+}
+
+class PieChartState extends State<EGPieChart> {
+  int touchedIndex = -1;
+
+  @override
+  Widget build(BuildContext context) {
+    return AspectRatio(
+      aspectRatio: 1,
+      child: Column(children: [
+        Row(
+          children: <Widget>[
+            Expanded(
+              child: AspectRatio(
+                aspectRatio: 1.3,
+                child: PieChart(
+                  PieChartData(
+                      pieTouchData: PieTouchData(touchCallback:
+                          (FlTouchEvent event, pieTouchResponse) {
+                        setState(() {
+                          if (!event.isInterestedForInteractions ||
+                              pieTouchResponse == null ||
+                              pieTouchResponse.touchedSection == null) {
+                            touchedIndex = -1;
+                            return;
```

Comment:
Nice null guard checks, I like that you return early if the event is invalid instead of 
```dart
if(valid) {
  // do stuff
}
```